array/array.go

```diff
@@ -0,0 +1,131 @@
+package array
+
+import (
+	"errors"
+)
+
+type Array struct {
+	arr  []int
```

Comment:
I think this property should not be named `arr`. We don't keep an array inside another array. `data` can be better name.

Reply:
Naming things is always difficult @@ I'll edit this.

Reply:
Another way is naming `arr` property as `items` so each value in array is an item.